src/integer/mat_z/sample/discrete_gauss.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains algorithms for sampling according to the discrete Gaussian distribution.
+
+use crate::{
+    error::MathError,
+    integer::{MatZ, Z},
+    rational::{MatQ, Q},
+    utils::sample::discrete_gauss::sample_d,
+};
+
+impl MatZ {
+    /// SampleD samples a discrete Gaussian from the lattice with `basis` as a subroutine.
```

Comment:
What happens if basis is not a basis?
In my opinion, just add a comment, that that is not checked for the corresponding functions